engine.py

```diff
@@ -132,7 +139,10 @@ def refresh_engine(engine_name, prev_context, menu_name):
     # API instance.
     try:
         tk = tank.tank_from_path(new_path)
+        logger.debug("Extracted sgtk instance: '%r' from path: '%r'" % (tk, new_path))
```

Comment:
For these debug logs where you're building strings, instead of using `logger.debug("...%r" % foobar)` you can save some processing time when debug logging is NOT on by doing the following: `logger.debug("...%r", foobar)`. This will stop the string interpolation from occurring if debug logging is off. The logger will do it for you instead, and will only do so if the logging severity threshold is met. Multiple insertions would work like this: `logger.debug("%r...%r", (foo, bar))`

Reply:
Worth noting that we really only need to worry about this for debug calls. Anything higher than that will always go to the logs, so there's no benefit to doing this for info, error, exception, etc in Toolkit code.

Reply:
I think the logging would need to be like this: `logger.debug("%r...%r", foo, bar)`, not `logger.debug("%r...%r", (foo, bar))`